### PR TITLE
SyliusReportBundle docs

### PR DESCRIPTION
**Pull request by Zales0123**

### CODE REVIEW

File: bundles/SyliusReportBundle/custom-data-fetcher.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+Adding custom data fetcher
+==========================
+
+SyliusReportBundle has some default data fetchers implemented, however, obviously they can be insufficient for some e-commerce systems. This chapter shows step-by-step way, how to add new data fetcher to make reports customization easier and more corresponding to your needs.
+
+Create custom data fetcher class
+--------------------------------
+
+First step is creation of our new data fetcher class. It should be placed in ``Sylius\Bundle\ReportBundle\DataFetcher`` namespace and to provide proper logic it has to implement ``Sylius\Component\Report\DataFetcher\DataFetcherInterface``. Because of implementation, data fetcher class must provide two methods:
```
Comment (michalmarcinkowski) [inline bundles/SyliusReportBundle/custom-data-fetcher.rst]: ``` It should be placed in... ``` - data fetcher class can be placed anywhere

File: components/Report/models.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Models
+======
+
+The Report
+-----------
+
+Report is the main model in SyliusReportComponent. This simple class represents every unique report in the system.
+The default interface contains the following attributes with appropriate setters and getters.
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: Interface does not have any attributes. ``` default model contains ``` sounds better.

File: components/Report/models.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+.. note::
+
+    This model implements ``ReportInterface``
+
+
+The Data
+-----------
+
+Data is supporting model in SyliusReportBundle. That contains data provided by data fetcher and labels used to clarify report display.
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: Component doesn't know anything about SyliusReportBundle. Data model represents report data in a uniform form.

File: components/Report/models.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
++--------------+------------------------------------+
+| Attribute    | Description                        |
++==============+====================================+
+| labels       | Labels that describe fetched data  |
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: ``` Labels that describe fetched data ``` => ``` Labels that describe data ``` - just data is enough

File: components/Report/models.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
++--------------+------------------------------------+
+| data         | Data fetcher from database         |
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: ``` array of values with data ``` - or something like that (it doesn't have to be fetched from database

File: components/Report/models.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+DataFetcherInterface
+----------------------
+
+To characterize data fetcher object, its class needs to implement the ``DataFetcherInterface``.
+
++-------------------------------+---------------------------------------+
+| Method                        | Description                           |
++===============================+=======================================+
+| fetch(array $configuration)   | Fetch data from database              |
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: Return data, based on given configuration

File: components/Report/models.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+RendererInterface
+----------------------
+
+To characterize renderer object, its class needs to implement the ``RendererInterface``.
+
++-----------------------------------------------+----------------------------------------------------------+
+| Method                                        | Description                                              |
++===============================================+==========================================================+
+| render(ReportInterface $report, Data $data)   | Render given data and report with propor configuration   |
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: ``` propor ``` => ``` proper ```

File: components/Report/models.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+----------------------
+
+To characterize data fetcher object, its class needs to implement the ``DataFetcherInterface``.
+
++-------------------------------+---------------------------------------+
+| Method                        | Description                           |
++===============================+=======================================+
+| fetch(array $configuration)   | Fetch data from database              |
++-------------------------------+---------------------------------------+
+| getType()                     | Return type of data fetcher object    |
```
Comment (michalmarcinkowski) [inline components/Report/models.rst]: We should stick to one form, IMO the description should be in the 3rd person (Returns, Renders etc.)

File: bundles/SyliusReportBundle/index.rst [hunk 9]
```diff
@@ -0,0 +1,15 @@
+SyliusReportBundle
+===================
+
+SyliusReportBundle is the flexible reports management system for Symfony2 e-commerce applications.
```
Comment (michalmarcinkowski) [inline bundles/SyliusReportBundle/index.rst]: is the flexible => is **a** flexible

for Symfony2 e-commerce applications => for Symfony2 ~~**e-commerce**~~ applications

File: bundles/SyliusReportBundle/index.rst [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+Using this bundle, you can generate various and complex reports for your webshop, starting from defaults - user registrations report, and total sales report. Implemented renderers allows you to display data in many different ways - from simple but clear table, to impressive and colorful charts.
```
Comment (michalmarcinkowski) [inline bundles/SyliusReportBundle/index.rst]: for your webshop => for your **system**

starting from defaults - user registrations report, and total sales report => starting from user registrations or total sales report (<link do data fetcherow>)

**Comment by pjedrzejewski** [comment]
Thanks Mateusz! :+1:

**Pull request merged**